### PR TITLE
SCAL-116040 host-event current iframe url

### PR DESCRIPTION
**Pull request by yinstardev**

### CODE REVIEW

File: src/utils/processTrigger.ts [hunk 1]
```diff
@@ -51,6 +60,9 @@ export function processTrigger(
             reload(iFrame);
             return res(null);
         }
+        if (messageType === HostEvent.GetIframeSrc) {
+            return getIframeSrc(iFrame);
```
Comment (sastaachar) [inline src/utils/processTrigger.ts]: pass the result into res check above

also test it out in app embed , if route changes return proper urls

File: src/utils/processTrigger.ts [hunk 2]
```diff
@@ -51,6 +60,9 @@ export function processTrigger(
             reload(iFrame);
             return res(null);
         }
+        if (messageType === HostEvent.GetIframeSrc) {
```
Comment (sastaachar) [inline src/utils/processTrigger.ts]: add tests

File: src/utils/processTrigger.ts (hunk 1 above)
Comment (yinstardev) [reply src/utils/processTrigger.ts]: yeah, right. 
Will test in app embed

File: src/utils/processTrigger.ts (hunk 2 above)
Comment (yinstardev) [reply src/utils/processTrigger.ts]: sure

File: src/types.ts [hunk 3]
```diff
@@ -2268,6 +2268,11 @@ export enum HostEvent {
      * @hidden
      */
     Reload = 'reload',
+    /**
+     * Get current iframe src
```
Comment (adityamittal3107) [inline src/types.ts:2272]: Add Examples and version

Comment (yinstardev) [reply src/types.ts:2272]: Added. Thanks

File: src/types.ts [hunk 4]
```diff
@@ -2268,6 +2268,16 @@ export enum HostEvent {
...
+     * @example
+     * ```js
+     * const frameUrl = AppEmbed.trigger(HostEvent.GetIframeUrl)
+     * ```
+     * @version SDK: 1.35.0 | Thoughtspot: 10.4.cl
+     * @hidden
```
Comment (adityamittal3107) [inline src/types.ts]: Why @hidden tag is added?

File: src/types.ts [hunk 5]
```diff
@@ -2268,6 +2268,16 @@ export enum HostEvent {
      * @hidden
      */
     Reload = 'reload',
+    /**
+     * Get current iframe src
+     * @example
+     * ```js
+     * const frameUrl = AppEmbed.trigger(HostEvent.GetIframeUrl)
+     * ```
+     * @version SDK: 1.35.0 | Thoughtspot: 10.4.cl
```
Comment (adityamittal3107) [inline src/types.ts]: Thoughtspot version should be 10.4.0.cl

Comment (yinstardev) [reply src/types.ts]: Thanks.

File: src/types.ts (hunk 4 above)
Comment (yinstardev) [reply src/types.ts]: Removed, It's not available in docs, will raise one ticket for @ShashiSubramanya

**Review by adityamittal3107 — APPROVED** [review]

**Pull request merged**